lib/db/mysql.js

```diff
@@ -1082,5 +1084,46 @@ module.exports = function (log, error) {
       )
   }
 
+  function ipHmac(key, uid, addr) {
+    if (ip.isV4Format(addr)) {
+      addr = '::' + addr
+    }
+    addr = ip.toBuffer(addr)
+
+    var hmac = crypto.createHmac('sha256', key)
+    hmac.update(uid)
+    hmac.update(addr)
+
+    return hmac.digest()
+  }
+
+  var SECURITY_EVENT_NAMES = {
+    'account.create': 1,
+    'account.login': 2,
+    'account.reset': 3
+  }
+
+  var CREATE_SECURITY_EVENT = 'CALL createSecurityEvent_1(?, ?, ?, ?, ?)'
+  MySql.prototype.createSecurityEvent = function (data) {
+    var uid = data.uid
+    var tokenId = data.tokenId
+    var nameId = SECURITY_EVENT_NAMES[data.name]
+    var ipAddr = ipHmac(this.ipHmacKey, uid, data.ipAddr)
+    return this.write(CREATE_SECURITY_EVENT, [uid, tokenId, nameId, ipAddr, Date.now()])
+  }
+
+  var FETCH_SECURITY_EVENTS = 'CALL fetchSecurityEvents_1(?, ?)'
+  MySql.prototype.securityEvents = function (where) {
+    var uid = where.uid
+
+    var ipAddr = ipHmac(this.ipHmacKey, uid, where.ipAddr)
```

Comment:
> We store the addresses as hmacs, so the auth-server cannot do the comparison
> itself unless we were to share the hmac key

As we iterate on this we might find that it's simpler for auth-server to old this key and do any hashing on its side before sending the values through to the db.  But I agree, it doesn't need to block the first version of this, and if we need to change it later it won't cost us much to do so.